src/reset/index.ts

```diff
@@ -1,15 +1,21 @@
-import { Unit, Store } from 'effector';
+import { createEvent } from 'effector';
 
-export function reset({
-  clock,
-  target,
-}: {
-  clock: Unit<any> | Array<Unit<any>>;
+import type { Event, Unit, Store } from 'effector';
+
+type Params = {
+  clock?: Unit<any> | Array<Unit<any>>;
   target: Store<any> | Array<Store<any>>;
-}): void {
+};
+
+export function reset(_: Required<Params>): void;
+export function reset(_: Pick<Params, 'target'>): Event<void>;
+
+export function reset({ clock, target }: Params) {
```

Comment:
```suggestion
export function reset({ clock = createEvent(), target }: Params) {
```

Reply:
I may be wrong, but we can't do that because checking `return clock ? clocks[0] : undefined` will never return `undefined`